### PR TITLE
Fix Gaussian elimination pivoting

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
### Describe your change:

Fix Gaussian elimination pivoting and warning:
```
linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py::linear_algebra.src.gaussian_elimination_pivoting.gaussian_elimination_pivoting.solve_linear_system
  /home/runner/work/Python/Python/linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:55: RuntimeWarning: invalid value encountered in scalar divide
    ab[i, column_num - 1]
```
on build `Run tests` stage

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by MaximSmolskiy** [comment]
@tianyizheng02 please review

File: linear_algebra/src/gaussian_elimination_pivoting.py [hunk 1]
```diff
@@ -22,40 +22,28 @@ def solve_linear_system(matrix: np.ndarray) -> np.ndarray:
...
             if abs(ab[i][column_num]) > abs(ab[column_num][column_num]):
                 ab[[column_num, i]] = ab[[i, column_num]]
-                if ab[column_num, column_num] == 0.0:
-                    raise ValueError("Matrix is not correct")
-            else:
-                pass
+
+        # Upper triangular matrix
+        if ab[column_num, column_num] == 0.0:
+            raise ValueError("Matrix is not correct")
```
[the remark this answers is not in the document]
Comment (MaximSmolskiy) [reply linear_algebra/src/gaussian_elimination_pivoting.py]: Done

File: linear_algebra/src/gaussian_elimination_pivoting.py [hunk 2]
```diff
@@ -22,40 +22,28 @@ def solve_linear_system(matrix: np.ndarray) -> np.ndarray:
...
-            else:
-                pass
-        if column_num != 0:
-            for i in range(column_num, num_of_rows):
-                ab[i, :] -= (
-                    ab[i, column_num - 1]
-                    / ab[column_num - 1, column_num - 1]
-                    * ab[column_num - 1, :]
-                )
+    assert num_of_rows == num_of_columns
```
[the remark this answers is not in the document]
Comment (MaximSmolskiy) [reply linear_algebra/src/gaussian_elimination_pivoting.py]: Done

File: linear_algebra/src/gaussian_elimination_pivoting.py [hunk 3]
```diff
@@ -22,40 +22,28 @@ def solve_linear_system(matrix: np.ndarray) -> np.ndarray:
...
         for i in range(column_num, num_of_columns):
             if abs(ab[i][column_num]) > abs(ab[column_num][column_num]):
                 ab[[column_num, i]] = ab[[i, column_num]]
-                if ab[column_num, column_num] == 0.0:
-                    raise ValueError("Matrix is not correct")
-            else:
-                pass
+
+        # Upper triangular matrix
+        if ab[column_num, column_num] == 0.0:
```
[the remark this answers is not in the document]
Comment (MaximSmolskiy) [reply linear_algebra/src/gaussian_elimination_pivoting.py]: Yes, with tolerance this comparison will be more reliable. Done

**Comment by MaximSmolskiy** [comment]
@tianyizheng02 please review

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**